src/Srclib/Types.hs

```diff
@@ -20,6 +24,121 @@ import Data.Text qualified as Text
 import Path (File, SomeBase)
 import Types (GraphBreadth (..))
 
+data LicenseSourceUnit = LicenseSourceUnit
+  { licenseSourceUnitName :: Text
+  , licenseSourceUnitType :: Text
+  , licenseSourceUnitLicenseUnits :: [LicenseUnit]
```

Comment:
You can get an empty `licenseSourceUnitLicenseUnits` if you run themis on an empty directory, so I left a comment explaining that

Reply:
Sorry, I should have been more clear - you can do something like:

```haskell 
licenseSourceUnitLicenseUnits :: Maybe (NonEmpty LicenseUnit)
```

This would be less error-prone, as it won't let you have empty [], and subsequent usage of this type by later code will need to explicitly deal with it being Nothing.

Reply:
Megh and I talked about this.

You can end up with a non-empty `[LicenseUnit]` if you scan an empty directory. I.e. if you point the vendored dep path at a directory that exists but doesn't have anything in it.

This actually works, in that it uploads an empty `[]` for the license scan result, and you end up with an unlicensed dependency issue in the UI.

This is technically correct, but is a really cryptic way of flagging "you tried to scan an empty directory".

So I've changed it like Megh suggested, and left a TODO to output a warning if we get `[]` back from Themis when scanning a directory